File: BankingApp_WDP.API/DTOs/Cards/CardResponse.cs
using BankingApp.Models;

namespace BankingApp.DTOs.Cards;

public class CardResponse
{
    public int Id { get; set; }
    public int AccountId { get; set; }
    public string MaskedCardNumber { get; set; } = string.Empty;
    public string ExpiryDate { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public decimal DailyLimit { get; set; }
    public decimal MonthlyLimit { get; set; }
    public DateTime CreatedAt { get; set; }

    public static CardResponse FromCard(Card card) => new()
    {
        Id = card.Id,
        AccountId = card.AccountId,
        MaskedCardNumber = MaskCardNumber(card.CardNumber),
        ExpiryDate = card.ExpiryDate.ToString("MM/yy"),
        Type = card.Type.ToString(),
        Status = card.Status.ToString(),
        DailyLimit = card.DailyLimit,
        MonthlyLimit = card.MonthlyLimit,
        CreatedAt = card.CreatedAt
    };

    private static string MaskCardNumber(string cardNumber)
    {
        var digits = cardNumber.Replace(" ", "");
        if (digits.Length < 4) return cardNumber;
        return "**** **** **** " + digits[^4..];
    }
}

File: BankingApp_WDP.APP/app/src/main/java/com/example/bankingapp/data/model/card/CardResponse.kt
package com.example.bankingapp.data.model.card

import com.google.gson.annotations.SerializedName

data class CardResponse(
    @SerializedName("id")               val id: String,
    @SerializedName("accountId")        val accountId: String,
    @SerializedName("maskedCardNumber") val maskedCardNumber: String,
    @SerializedName("expiryDate")       val expiryDate: String,
    @SerializedName("type")             val type: String,
    @SerializedName("status")           val status: String,
    @SerializedName("dailyLimit")       val dailyLimit: Double,
    @SerializedName("monthlyLimit")     val monthlyLimit: Double,
    @SerializedName("createdAt")        val createdAt: String
)
